src/flow_export.cc

```diff
@@ -20,6 +21,43 @@ tpRank( struct ncclComm & comm, int rank ) {
    return rank == -1 ? UINT32_MAX : comm.topParentRanks[ rank ];
 }
 
+inline void
+initFlowV1( const connectionMetaData_t & cmd, Flow_v1 & flow ) {
+   flow.commHash = currCommHash[ cmd.tpLocalRank ];
+   flow.direction = cmd.send ? SEND : RECEIVE;
+   flow.srcRank = cmd.tpRank;
+   flow.srcLocalRank = cmd.tpLocalRank;
+   flow.dstRank = cmd.tpRemoteRank;
+   flow.channel = cmd.channelId;
+   flow.topology = cmd.connIndex == 0 ? COLLECTIVE : P2P;
+}
+
+inline void
+gidToIpAddr( const union ibv_gid & gid, IpGenAddr & genAddr ) {
+   static const uint8_t ipv4Prefix[ 12 ] = {
+      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0xff, 0xff
+   };
+   if ( gid.global.subnet_prefix == 0 && memcmp( gid.raw, ipv4Prefix, 12 ) == 0 ) {
```

Comment:
It would be good to add a comment for the constant values like (12/16)

Reply:
Done. Added a comment in commit 87f1260c09ed89c342bbba3dea8ce949718a4726.